.forge-snapshots/set protocol fee.snap

```diff
@@ -0,0 +1 @@
+32422
```

Comment:
Can u do a comparison from main to see how much this improves gas?

Reply:
same with calculate swap fee?

Reply:
The gas improvement from optimizing `validate` can be seen from this [commit](https://github.com/Uniswap/v4-core/pull/283/commits/43033f3819d8ac1b645777e138ea2ecf50982180). The "set protocol fee" snapshot on the current main is actually 32519.
The performance gain on `calculateSwapFee` can be seen [here](https://github.com/Uniswap/v4-core/pull/283/commits/6632257dc90e94ffe83309c5ac5a174b0699e02f).
This is the forge snapshot diff result.
```
test_calculateSwapFee() (gas: -495 (-9.192%)) 
```